lib/palindrome_check.rb

```diff
@@ -1,5 +1,33 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  end
+
+  low_index = 0
+  high_index = (my_phrase.length - 1)
+
+  while low_index <= high_index
+    while my_phrase[low_index] == " "
```

Comment:
In the two nested inner loops, continue to check and confirm that low_index is less than high_index before comparing the character at the index low_index (or high_index)  with a white space.